sql/parser/builtins.go

```diff
@@ -1161,6 +1186,31 @@ func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
 	return fmt.Sprintf("(?%s:%s)", flagString.String(), pattern), nil
 }
 
+func overlay(s, to string, pos, size int) (Datum, error) {
+	if pos < 1 {
+		return nil, errors.New("negative substring length not allowed")
+	}
+
+	var out bytes.Buffer
+	i := 0
+	for _, c := range s {
+		if i >= pos-1 {
+			break
+		}
+		out.WriteRune(c)
+		i++
+	}
```

Comment:
Yeah, I was just looking for alternatives that didn't require a buffer copy, but I guess we can look into optimizing that later.